bitcoin/src/network.rs

```diff
@@ -170,14 +234,14 @@ impl Network {
     }
 
     /// Returns the associated network parameters.
-    pub const fn params(self) -> &'static Params {
-        const PARAMS: [Params; 4] = [
-            Params::new(Network::Bitcoin),
-            Params::new(Network::Testnet),
-            Params::new(Network::Signet),
-            Params::new(Network::Regtest),
-        ];
-        &PARAMS[self as usize]
+    pub fn params(self) -> &'static Params {
```

Comment:
```suggestion
    pub const fn params(self) -> &'static Params {
```

I see no reason to remove the `const` keyword here, the method still works at compile time.